### PR TITLE
Remove livecheck failures from autobump list

### PR DESCRIPTION
**Pull request by carlocab**
The livecheck blocks for autobumped formulae can silently fail. Let's
handle that by automatically removing them from the autobump list when
that happens.

### CODE REVIEW

File: .github/workflows/livecheck.yml [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+      - name: Open pull request
+        if: fromJSON(steps.commit.outputs.push)
+        env:
+          GITHUB_TOKEN: ${{ secrets.HOMEBREW_GITHUB_PUBLIC_REPO_TOKEN }}
+        run: |
+          if ! jq --exit-status '.state == "OPEN"'
+          then
+            gh pr create --fill
+          fi < <(gh pr view --json=state)
```
Comment (ZhongRuoyu) [inline .github/workflows/livecheck.yml:185]: `gh pr view` does not work well when there are multiple PRs with the same branch name. (Try to checkout a new branch named `sponsors-maintainers-man-completions` in `brew --repo` and see.) The `sync-shared-config` workflow [relies on the API](https://github.com/Homebrew/.github/blob/c65c8b0c816d261969bfd897caff6bb472f82ff7/.github/workflows/sync-shared-config.yml#L113-L129) for this reason.

File: .github/workflows/livecheck.yml [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        env:
+          GIT_COMMITTER_NAME: ${{ steps.git-user-config.outputs.name }}
+          GIT_COMMITTER_EMAIL: ${{ steps.git-user-config.outputs.email }}
+          HOMEBREW_GPG_PASSPHRASE: ${{ secrets.BREWTESTBOT_GPG_SIGNING_SUBKEY_PASSPHRASE }}
+
+      - name: Open pull request
+        if: fromJSON(steps.commit.outputs.push)
+        env:
+          GITHUB_TOKEN: ${{ secrets.HOMEBREW_GITHUB_PUBLIC_REPO_TOKEN }}
+        run: |
```
Comment (ZhongRuoyu) [inline .github/workflows/livecheck.yml:189]:
```suggestion
        working-directory: ${{ steps.set-up-homebrew.outputs.repository-path }}
        run: |
```

**Review by MikeMcQuaid — COMMENTED** [review]
Looks good so far! Could this be part of the existing autobump workflow instead? Feels like having another workflow essentially doing autobump but not using the results is going to burn through our rate limits.

**Comment by carlocab** [comment]
> Looks good so far! Could this be part of the existing autobump workflow instead?

It could, but,

> Feels like having another workflow essentially doing autobump but not using the results is going to burn through our rate limits.

I don't think it will help with this, since it'll likely have to do `brew livecheck` separately and then throw away the result by doing `brew bump` anyway regardless. Will have a look.

**Comment by MikeMcQuaid** [comment]
> I don't think it will help with this, since it'll likely have to do `brew livecheck` separately and then throw away the result by doing `brew bump` anyway regardless.

I'm thinking that we could have something in e.g. `brew bump` itself to manage the autobump list and create PRs directly from that?

Would also be nice to have more logic in Homebrew/brew and less in (more) workflows if possible.

**Comment by carlocab** [comment]
> I'm thinking that we could have something in e.g. `brew bump` itself to manage the autobump list and create PRs directly from that?

Yea, maybe. `brew bump` would need to know about the autobump list, though.

Here's a start: https://github.com/Homebrew/brew/pull/18227

It then needs to handle removal from the autobump list when a step fails, but `bump.rb` isn't really very transparent about where stuff fails.

**Pull request closed without merging**